### PR TITLE
add new board: splaytoraid

### PR DESCRIPTION
**Pull request by Conor-Burns**
## Description
adding splaytoraid a 40% ergonomic keyboard with a stacked acrylic case and RGB underglow.

splay

## Types of Changes
- add new board with two subfolder for different mcu's

- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/0xcb/splaytoraid/info.json [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    "manufacturer": "Freya",
+    "keyboard_name": "splaytoraid",
+    "maintainer": "freya-irl",
+    "url": "https://github.com/freya-irl/splaytoraid40",
+    "debounce": 5,
+    "diode_direction": "COL2ROW",
+    "matrix_size": {
+        "cols": 7,
+        "rows": 8
+    },
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/info.json:11]:
```suggestion
```

File: keyboards/0xcb/splaytoraid/info.json [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0xCB00",
+        "vid": "0x2004"
+    },
+    "features": {
+        "encoder_map": true,
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/info.json:18]: This feature should not be enabled at the keyboard level, and needs to be moved to `keymaps/*/rules.mk`.
```suggestion
```

File: keyboards/0xcb/splaytoraid/keymaps/pi/config.h [hunk 3]
```diff
@@ -0,0 +1,87 @@
+#pragma once
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/keymaps/pi/config.h:1]: This file is missing a license header.

File: keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c [hunk 4]
```diff
@@ -0,0 +1,415 @@
+#include QMK_KEYBOARD_H
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c:1]: This file is missing a license header.

File: keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c [hunk 5]
```diff
@@ -0,0 +1,415 @@
...
+#include <stdio.h>
+#include <string.h>
+#include "splaytoraid40.h"
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c:4]: This file does not exist.
```suggestion
```

File: keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c [hunk 6]
```diff
@@ -0,0 +1,415 @@
...
+   │    Q    │    W    │    E    │    R    │    T    │ ╭╮╭╮╭╮╭╮╭╮╭╮╭╮╭╮╭╮ │    Y    │    U    │    I    │    O    │    P    │
+   ├─────────┼─────────┼─────────┼─────────┼─────────┤ │╰╯╰╯╰╯╰╯╰╯╰╯╰╯╰╯│ ├─────────┼─────────┼─────────┼─────────┼─────────┤
+   │    A    │    S    │    D    │    F    │    G    ├─╯                ╰─┤    H    │    J    │    K    │    L    │    ;    │
+   ├─────────┼─────────┼─────────┼─────────┼─────────┤╭────────╮╭────────╮├─────────┼─────────┼─────────┼─────────┼─────────┤
+   │    Z    │    X    │    C    │    V    │    B    ││        ││        ││    N    │    M    │    ,    │    .    │    /    │
+   └─────────┴─────────┼─────────┼─────────┼─────────┼╰────────╯╰────────╯┼─────────┼─────────┼─────────┼─────────┴─────────┘
+                       │   ESC   │   TAB   │  ENTER  │                    │  SPACE  │ BSPACE  │   DEL   │
+                       └─────────┴─────────┴─────────┘                    └─────────┴─────────┴─────────┘ */
+
+   [_QWERTY] = LAYOUT_splaytoraid36(
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c:102]: This layout does not exist.

File: keyboards/0xcb/splaytoraid/keymaps/pi/layers.h [hunk 7]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+// layer enum moved here so it can also be used by other files
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/keymaps/pi/layers.h:3]: These aren't used anywhere else, and should be moved to the keymap (and this file removed)

File: keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c [hunk 8]
```diff
@@ -0,0 +1,415 @@
...
+    switch (keycode) {
+        // case SHT_T:
+        //     return TAPPING_TERM - 150; // Recommended
+        // case SHT_N:
+        //     return TAPPING_TERM - 150; // Recommended
+        default:
+            return TAPPING_TERM;
+    }
+}
+
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c:268]:
```suggestion
```

File: keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c [hunk 9]
```diff
@@ -0,0 +1,415 @@
...
+// └────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+// ▝▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▘
+void keyboard_post_init_user(void) {
+  // Call the post init code.
+  #if RGB_MATRIX_ENABLE
+    rgblight_enable_noeeprom();
+    //rgblight_sethsv_noeeprom(35, 255, 255); // set default RGB color to yellow
+  #endif //RGB_MATRIX_ENABLE
+}
+
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c:302]: This doesn't really do anything as `RGB_MATRIX` is disabled, at it's calling an `RGBLIGHT` specific function...
```suggestion
```

File: keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c [hunk 10]
```diff
@@ -0,0 +1,415 @@
...
+                }
+              #ifdef HAPTIC_ENABLE
+                DRV_pulse(pulsing_strong);
+              #endif // HAPTIC_ENABLE
+            eeconfig_update_keymap(keymap_config.raw);
+            clear_keyboard();  // ──── clear to prevent stuck keys
+            return false;
+          }
+
+
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c:332]:
```suggestion
```

File: keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c [hunk 11]
```diff
@@ -0,0 +1,415 @@
...
+                  DRV_pulse(transition_hum);
+                #endif // HAPTIC_ENABLE
+            }
+            return false;
+        case QWERTY:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_QWERTY);
+                #ifdef HAPTIC_ENABLE
+                  DRV_pulse(transition_hum);
+                #endif // HAPTIC_ENABLE
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c:350]:
```suggestion
            }
            return false;
        case QWERTY:
            if (record->event.pressed) {
                set_single_persistent_default_layer(_QWERTY);
```

File: keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c [hunk 12]
```diff
@@ -0,0 +1,415 @@
...
+            if(active) { tap_code(KC_MUTE); }
+            break;
+    }
+    return true;
+}
+#endif //
+
+
+
+
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c:415]:
```suggestion
```

File: keyboards/0xcb/splaytoraid/keymaps/vial/config.h [hunk 13]
```diff
@@ -0,0 +1,9 @@
+// Copyright 2023 Conor Burns (@Conor-Burns)
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/keymaps/vial/config.h:1]: This directory should be removed. Vial keymaps are not accepted.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/0xcb/splaytoraid/info.json (hunk 1 above)
Comment (Conor-Burns) [reply keyboards/0xcb/splaytoraid/info.json:11]: what's wrong here?

**Comment by Conor-Burns** [comment]
@waffle87 thank you for your time reviewing this! I've commited the changes apart from the first one - that won't work right?

File: keyboards/0xcb/splaytoraid/info.json (hunk 1 above)
Comment (Conor-Burns) [reply keyboards/0xcb/splaytoraid/info.json:11]: ohhh awesome :D

File: keyboards/0xcb/splaytoraid/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+              _______,  _______,  _______,  _______,  _______,            _______,  _______,  _______,  _______,  _______,
+                                  _______,  _______,  _______,  _______,  _______,  _______,  _______
+  ),
+
+  [_ADJUST] = LAYOUT_40(
+    _______,  _______,  _______,  _______,  _______,  _______,            _______,  _______,  _______,  _______,  _______,  _______,
+    _______,  _______,  _______,  _______,  _______,  _______,            _______,  _______,  _______,  _______,  _______,  _______,
+              _______,  _______,  _______,  _______,  _______,            _______,  _______,  _______,  _______,  _______,
+                                  _______,  _______,  _______,  _______,  _______,  _______,  _______
+  ),
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/keymaps/default/keymap.c:40]:
```suggestion
  )
```

File: keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c [hunk 15]
```diff
@@ -0,0 +1,350 @@
...
+                  #ifdef AUDIO_ENABLE
+                    PLAY_SONG(winxp_song);
+                  #endif // AUDIO_ENABLE
+                }
+            eeconfig_update_keymap(keymap_config.raw);
+            clear_keyboard();  // ──── clear to prevent stuck keys
+            return false;
+          }
+
+
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c:316]: [magic](https://docs.qmk.fm/#/keycodes_magic) keycodes should be used here instead, as it seems this keyboard does not have audio support, and this keymap was entirely copy/pasted from another keyboard.
```suggestion
```

File: keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c [hunk 16]
```diff
@@ -0,0 +1,350 @@
...
+};
+
+// ┌───────────────────────────────────────────────────────────┐
+// │ d e f i n e   k e y c o d e s                             │
+// └───────────────────────────────────────────────────────────┘
+
+enum custom_keycodes {
+    COLEMAK = SAFE_RANGE,
+    QWERTY,
+    OS_SWAP
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/keymaps/pi/keymap.c:29]:
```suggestion
    COLEMAK = QK_USER,
    QWERTY
```

File: keyboards/0xcb/splaytoraid/readme.md [hunk 17]
```diff
@@ -0,0 +1,37 @@
+# splaytoraid
+
+![splaytoraid](https://keeb.supply/assets/splaytoraid-rgb__preview.webp?preset=full)
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/readme.md:3]: This image should be rehosted on Imgur.

File: keyboards/0xcb/splaytoraid/readme.md [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+    make 0xcb/splaytoraid:default
+
+this will build the rp2040 variant
+you can also build the atmega32u4 variant by using:
+
+    make 0xcb/splaytoraid/32u4:default
+    
+and flash it:
+
+    make 0xcb/splaytoraid/32u4:default:flash
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/readme.md:26]:
```suggestion
    make 0xcb/splaytoraid:default
    make 0xcb/splaytoraid/32u4:default

Flashing example for this keyboard:

    make 0xcb/splaytoraid:default:flash
    make 0xcb/splaytoraid/32u4:default:flash
```

File: keyboards/0xcb/splaytoraid/rp2040/config.h [hunk 19]
```diff
@@ -0,0 +1,31 @@
+// Copyright 2023 Conor Burns (@Conor-Burns)
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#pragma once
+
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/rp2040/config.h:5]:
```suggestion
#define RGB_DI_PIN GP3
```

File: keyboards/0xcb/splaytoraid/32u4/info.json [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+            {
+                "pin_a": "B5",
+                "pin_b": "B6",
+                "resolution": 4
+            }
+        ]
+    },
+    "rgblight": {
+        "pin": "D0"
+    }
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/32u4/info.json:23]: `RGBLIGHT` configuration shouldn't be used when the keyboard is meant to use `RGB_MATRIX` — so this should be placed in a `config.h` file.
```suggestion
```

File: keyboards/0xcb/splaytoraid/rp2040/info.json [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+            {
+                "pin_a": "GP9",
+                "pin_b": "GP21",
+                "resolution": 4
+            }
+        ]
+    },
+    "rgblight": {
+        "pin": "GP3"
+    }
```
Comment (waffle87) [inline keyboards/0xcb/splaytoraid/rp2040/info.json:20]:
```suggestion
    }
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Review by waffle87 — APPROVED** [review]

**Comment by Conor-Burns** [comment]
@lesshonor thank you for fixing the json! I'm a bit split about just using the pro micro pin names when I can avoid it - what do you think about the solution I just pushed?

**Comment by Conor-Burns** [comment]
@lesshonor thanks for explaining - you convinced me. I've added the 32u4 pins to the root json.

**Review by keyboard-magpie — APPROVED** [review]

**Pull request merged**